### DISCUSSION TITLE
Import.spec changes since version 0.31.0

### DISCUSSION THREAD

**Discussion by DimitarStefanovMihov** [vmware/govmomi#3244] [Community Help] [answered]
I appologies for the simplistic question, but before this version the output from ``` govc import.spec ``` was something like this: 
    {
      "Key": "dns_servers",
      "Value": ""
    },
    {
      "Key": "ntp_servers",
      "Value": ""
    }
  ],
  "NetworkMapping": [
    {
      "Name": "net0",
      "Network": ""
    },
    {
      "Name": "net1",
      "Network": ""
    }

but now the ``` Key ``` and ``` Value ``` in the PropertyMapping are lowercase and the rest for example in NetworkMapping sections are still uppercase:

    {
      "key": "dns_servers",
      "value": ""
    },
    {
      "key": "ntp_servers",
      "value": ""
    }
  ],
  "NetworkMapping": [
    {
      "Name": "net0",
      "Network": ""
    },

Is this something that was changed on purpose ? Thank you for your time.

**Comment by dougm** [comment]
Sorry for the confusion. We did call out in the [release notes](https://github.com/vmware/govmomi/releases/tag/v0.31.0):
> Updating the vim25/types with camelCased, JSON tags affects the output of the govc command when using the -json flag to emit results as JSON

This only applies to the generated vim25/types. The majority of the import structure is not generated, but hand written: https://github.com/vmware/govmomi/blob/2fab8d5337155ac7e6162b4cddd0b87fbfdf7515/govc/importx/options.go#L53-L59

With the Property.{key,value} fields being one of the generated vim25/types, `types.KeyValue`: https://github.com/vmware/govmomi/blob/2fab8d5337155ac7e6162b4cddd0b87fbfdf7515/govc/importx/options.go#L32-L33

The case change for import.ova input here is likely to cause more confusion that case change with output. I'll take a look at implementing a case-insensitive decoder for Property.{key,value} and we could release that with a v0.31.1

cc @akutz

**Reply by dougm** [reply, marked as answer]
Fixed in https://github.com/vmware/govmomi/releases/tag/v0.32.0
govc import.spec will output the same case as releases prior to 0.31, but with -options input 'key' and 'value' can be lower case.
May look at making the entire spec case-insensitive in the future, but for now existing spec json will behave as in the past.
